File: spojPractice/spoj1025.cpp
#include <iostream>
#include <algorithm>
using namespace std;
void merge(int *a, int low, int high, int mid)
{
    int i, j, k, c[50];
    i = low;
    k = low;
    j = mid + 1;
    while (i <= mid && j <= high)
    {
        if (a[i] < a[j])
        {
            c[k] = a[i];
            k++;
            i++;
        }
        else
        {
            c[k] = a[j];
            k++;
            j++;
        }
    }
    while (i <= mid)
    {
        c[k] = a[i];
        k++;
        i++;
    }
    while (j <= high)
    {
        c[k] = a[j];
        k++;
        j++;
    }
    for (i = low; i < k; i++)
    {
        a[i] = c[i];
    }
}
void mergesort(int *a, int low, int high)
{
    int mid;
    if (low < high)
    {
        mid=(low+high)/2;
        mergesort(a,low,mid);
        mergesort(a,mid+1,high);
        merge(a,low,high,mid);
    }
    return;
}
int main()
{
	int t;
	cin>>t;
	for(int q=1;q<=t;q++)
	{
		int n;
		cin>>n;
		int *m;
		int *w;
		m=new int[n];
		w=new int[n];
		for(int i=0;i<n;i++)
			cin>>m[i];
		for(int i=0;i<n;i++)
			cin>>w[i];
		sort(m,m+n);
		sort(w,w+n);
		int sum=0;
		for(int i=0;i<n;i++)
			sum+=m[i]*w[i];
		cout<<sum<<endl;
	}
}

File: spojPractice/spoj1025.java
import java.io.*;
class spoj1025
{
	public static void main(String args[])throws IOException
	{
		BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
		int t=Integer.parseInt(br.readLine());
		for(int q=1;q<=t;q++)
		{
			int n=Integer.parseInt(br.readLine());
			int m[]=new int[n];
			int w[]=new int[n];
			String lines=br.readLine();
			String[] strs=lines.trim().split("\\s+");
			for(int i=0;i<n;i++)
			{
				m[i]=Integer.parseInt(strs[i]);
			}
			String lines1=br.readLine();
			String[] strs1=lines1.trim().split("\\s+");
			for(int i=0;i<n;i++)
			{
				w[i]=Integer.parseInt(strs1[i]);
			}
			spoj1025 obj=new spoj1025();
			m=obj.mergeSort(m);
			w=obj.mergeSort(w);
			int sum=0;
			for(int i=0;i<n;i++)
			{
				sum+=m[i]*w[i];
			}
			System.out.println(sum);
		}
	}
	public int[] mergeSort(int [] a)
    {
        if(a.length<=1)
        {
            return a;
        }
        int firstLength=a.length/2;
        int secondLength=a.length-firstLength;
        int first[]=new int[firstLength];
        int second[]=new int[secondLength];
        System.arraycopy(a,0,first,0,firstLength);
        System.arraycopy(a,firstLength,second,0,secondLength);
        first=mergeSort(first);
        second=mergeSort(second);
        merge(first,second,a);
        return a;
    } 

    public void merge(int [] first,int [] second,int [] a)
    {
        int iFirst=0;
        int iSecond=0;
        int j=0;
        while(iFirst<first.length&&iSecond<second.length)
        {
            if(first[iFirst]<second[iSecond])
            {
                a[j]=first[iFirst];
                iFirst++;
            }
            else
            {
                a[j]=second[iSecond];
                iSecond++;
            }
            j++;
        }
        System.arraycopy(first,iFirst,a,j,first.length-iFirst);
        System.arraycopy(second,iSecond,a,j,second.length-iSecond);
    }
}